### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mat-hek**

### CODE REVIEW

File: lib/membrane/rtp/payload_format.ex [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    Application.put_env(@app, {:format, encoding_name}, payload_format)
+  end
+
+  defp merge_format(_name, _k, nil, v), do: v
+  defp merge_format(_name, _k, v, nil), do: v
+  defp merge_format(_name, _k, v, v), do: v
+
+  defp merge_format(name, k, v1, v2) do
+    raise "Cannot register RTP payload format #{name} field #{k} to #{inspect(v2)}, " <>
```
Comment (mickel8) [inline lib/membrane/rtp/payload_format.ex:125]: Shouldn't be here
```suggestion
    raise "Cannot register RTP payload format #{name} field #{k} in #{inspect(v2)}, " <>
```
?

File: lib/membrane/rtp/payload_format.ex [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        })
+    end
+  end
+
+  @doc """
+  Returns payload format registered for given encoding name.
+  """
+  @spec get(RTP.encoding_name_t()) :: t
+  def get(encoding_name) do
+    Application.get_env(@app, {:format, encoding_name}, %__MODULE__{encoding_name: encoding_name})
```
Comment (mickel8) [inline lib/membrane/rtp/payload_format.ex:103]: Shouldn't we return here empty map or nil? I mean doc says `registered` but if get_env returns default value it means given `encoding_name` isn't registered so we probably should return `%{}/nil`

File: lib/membrane/rtp/payload_format.ex [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    32 => %{encoding_name: :MPV, clock_rate: 90000},
+    33 => %{encoding_name: :MP2T, clock_rate: 90000},
+    34 => %{encoding_name: :H263, clock_rate: 90000}
+  }
+
+  @enforce_keys [:encoding_name]
+  defstruct @enforce_keys ++ [payload_type: nil, payloader: nil, depayloader: nil]
+
+  @type t :: %__MODULE__{
+          encoding_name: RTP.encoding_name_t(),
```
Comment (mickel8) [inline lib/membrane/rtp/payload_format.ex:44]: Do we need this field? If I understand it properly we do two types of registration:

`{:format, encoding_name} => %__MODULE__{}` - in this case we don't need encoding name in struct
and 
`{:payload_type, payload_type} => {encoding_name, clock_rate}` - here we don't use struct

File: lib/membrane/rtp/payload_format.ex [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  Registers default encoding name and clock rate for a dynamic payload_type
+  """
+  @spec register_payload_type_mapping(
+          RTP.dynamic_payload_type_t(),
+          RTP.encoding_name_t(),
+          RTP.clock_rate_t()
+        ) :: :ok | no_return()
+  def register_payload_type_mapping(payload_type, encoding_name, clock_rate)
+      when RTP.is_payload_type_dynamic(payload_type) do
+    case Application.fetch_env(@app, {:payload_type, payload_type}) do
```
Comment (mickel8) [inline lib/membrane/rtp/payload_format.ex:86]: I am not sure if this follows the convention from `{:format, encoding_name}` in which we put `format` under `encoding_name`. Here we put `clock_rate` and `encoding_name` under `payload_type` but we write `{:payload_type, payload_type}`. Maybe rename it to sth like `{:encoding_name_clock_rate, payload_type}`?

**Review by mickel8 — CHANGES_REQUESTED** [review]
I am wondering if we shouldn't do only one registration like `{:format, encoding_name} => %__MODULE__{}` and store the clock_rate in this struct but there are probably some use cases in which we also want `{:payload_type, payload_type} => {encoding_name, clock_rate}`

File: lib/membrane/rtp/payload_format.ex (hunk 2 above)
Comment (mat-hek) [reply lib/membrane/rtp/payload_format.ex:103]: The format can be also 'partially registered', for example, there's a payloader but not a depayloader. In other words, unknown fields are always set to `nil`. Thus, returning some custom value for unregistered format would make that inconvenient to interact with.

File: lib/membrane/rtp/payload_format.ex (hunk 3 above)
Comment (mat-hek) [reply lib/membrane/rtp/payload_format.ex:44]: I don't insist on this form, but here's the intention: we have a _set_ of registered formats, that should be searchable via encoding_name, and _mapping_ from payload type to some parameters. The mapping is intended to be used just to get assigned value, while the set of formats may be used to obtain the format and pass it somewhere

**Comment by mat-hek** [comment]
> I am wondering if we shouldn't do only one registration like `{:format, encoding_name} => %__MODULE__{}` and store the clock_rate in this struct but there are probably some use cases in which we also want `{:payload_type, payload_type} => {encoding_name, clock_rate}`

@mickel8 there are, some encodings have multiple corresponding payload types, with different clock rates

**Review by mickel8 — COMMENTED** [review]
I am wondering about one thing. `payload_type` under which we register `encoding_name` and `clock_rate` and which is present in `PayloadFormat`  struct isn't known at the time of registration. We will get it after SDP Offer. Unless we are offering side then it is probably ok. Maybef we should pass this `payload_type` as an argument and change time of registration if this is possible at all.

**Comment by mat-hek** [comment]
> I am wondering about one thing. `payload_type` under which we register `encoding_name` and `clock_rate` and which is present in `PayloadFormat` struct isn't known at the time of registration. We will get it after SDP Offer. Unless we are offering side then it is probably ok. Maybef we should pass this `payload_type` as an argument and change time of registration if this is possible at all.

@mickel8 this is the default payload type, usually specified in RFC describing how to place that given in RTP. If you'd like to use different payload type, you should pass it explicitly to whatever needs it.

**Review by mickel8 — APPROVED** [review]

**Pull request merged**